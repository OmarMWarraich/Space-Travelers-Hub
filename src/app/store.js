import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
/* eslint-disable-next-line import/prefer-default-export */
export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});
