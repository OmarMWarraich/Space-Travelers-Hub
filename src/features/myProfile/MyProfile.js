import React from 'react';
import { useSelector } from 'react-redux';
import styles from './MyProfile.module.css';
import Card from './Card';

const MyProfile = () => {
  const RocketsList = useSelector((state) => state.rockets);

  return (
    <>
      <div className={styles.myProfile}>
        <div className="col-md-6">
          <div className="card">
            <div className="card-body">
              <h5 className="card-title">My Profile</h5>
              <hr />
              <p className="card-text">Name: </p>
              <hr />
              <p className="card-text">Email: </p>
              <hr />
              <p className="card-text">Phone: </p>
            </div>
          </div>
        </div>

        <div className="col-md-6">
          <div className="card">
            <div className="card-body">
              <h5 className="card-title">My Rockets</h5>
              <hr />
              {RocketsList.filter((rocket) => rocket.reserved).map((rocket) => (
                <Card key={rocket.id} name={rocket.name} />))}
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default MyProfile;
