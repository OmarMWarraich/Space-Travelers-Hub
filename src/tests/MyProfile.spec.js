import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import MyProfile from '../features/myProfile/MyProfile';

it('MyProfile page rendering', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
