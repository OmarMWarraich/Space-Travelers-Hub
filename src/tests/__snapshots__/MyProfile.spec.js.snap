// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile page rendering 1`] = `
<div
  className="myProfile"
>
  <div
    className="col-md-6"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <h1
          className="cardTitle"
        >
          My Missions
        </h1>
        <hr />
        <h3
          className="text-center"
        >
          No Missions Joined
        </h3>
        <hr />
      </div>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <h1
          className="cardTitle"
        >
          My Rockets
        </h1>
        <hr />
        <h3
          className="text-center"
        >
          No Rockets Joined
        </h3>
        <hr />
      </div>
    </div>
  </div>
</div>
`;
